lib/wasi/src/syscalls/mod.rs

```diff
@@ -76,16 +75,18 @@ fn write_bytes<T: Write>(
 fn read_bytes<T: Read>(
     mut reader: T,
     memory: &Memory,
-    iovs_arr_cell: &[Cell<__wasi_iovec_t>],
+    iovs_arr_cell: Vec<WasmCell<__wasi_iovec_t>>,
```

Comment:
This should take a `&[]` to be more generic, it's not relevant to this function that the data is  stored in a `Vec`

Reply:
Fixed